source/xep/httpFileUpload/HttpFileUploadManager.cpp

```diff
@@ -46,13 +47,17 @@ bool HttpFileUploadManager::requestToUploadFileForJid(const QString &file, const
     if (busy_ == false && client_ != nullptr && serverHasFeatureHttpUpload_ == true)
     {
         QFile inputFile(file);
-        QString fileToUpload = createTargetImageName(file);
+        QString fileToUpload = createTargetFileName(file);
 
-        // don't resize image if server can handle it
-        if(inputFile.size() < getMaxFileSize()) 
+        fileType_ = QMimeDatabase().mimeTypeForFile(file).name();
+
+        // Resize image if server cannot handle it or if requested
+        if(fileType_.startsWith("image") && compressImages_)
+            returnValue = ImageProcessing::prepareImageForSending(file, fileToUpload, qMin(getMaxFileSize(), 400000u));
```

Comment:
Why do you limit this to 400000u?

Reply:
Compressing images makes sense when transfering the original file is not required. It keeps small files, easier to display. I was looking for a good threshold and not using the maximum value for file uploads which can be quite high. This value gives image file sizes between 100K and 400K.

Reply:
Ok. But what if the server signals to be able to handle larger attachments then the 400K? The server can fulfill the functionality but the code limits it? Are you concerned about performance issues of shrinking down the image size on large files?

Reply:
Yes I'm concerned by performance issues but also by people who have limited data plan with their mobile phone.

Reply:
This is an individual setting. It should not be strictly limited. If you want to do so, please make this configurable in the settings page.

Reply:
I agree. I will add it.